src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/PackageRestoreProgressTracker.PackageRestoreProgressTrackerInstance.cs

```diff
@@ -87,15 +91,20 @@ protected override Task InitializeCoreAsync(CancellationToken cancellationToken)
                         linkOptions: DataflowOption.PropagateCompletion,
                         cancellationToken: cancellationToken);
 
+                _packageReferenceTelemetryService.PostPackageRestoreEvent(PackageRestoreOperationNames.PackageRestoreProgressTrackerInstanceInitialized, ConfiguredProject.UnconfiguredProject.FullPath);
+
                 return Task.CompletedTask;
             }
 
             internal void OnRestoreCompleted(IProjectVersionedValue<ValueTuple<IProjectSnapshot, RestoreData>> value)
```

Comment:
Nit: Might be best to replace `ValueTuple` with named ValueTuple via the `( )` syntax as you run into them. Otherwise, you end up with the calls below like `value.Value.Item1` which is not easily read on first glance.

Reply:
makes sense. I will change that in a follow-up PR though.

---
In reply to: [567164943](https://github.com/dotnet/project-system/pull/6880#discussion_r567164943) [](ancestors = 567164943)